packages/lists/src/Queue.js

```diff
@@ -22,6 +26,25 @@ class Queue {
    *  @spaceComplexity O(1)
    */
   enqueue(item) {
+    if (!item) {
+      throw `${item} is not valid`;
+    }
+
+    const newNode = {
+      item: item,
+      points: null,
+    };
+
+   if (this.isEmpty()) {
+      this._head = newNode
+      this._tail = this._head;
+    } else { //this I don't fully gronk
```

Comment:
http://www.catb.org/jargon/html/G/grok.html Just so you know why I keep using this word.